cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/data/CassandraDataLayer.java

```diff
@@ -639,6 +644,16 @@ public CassandraRing createCassandraRingFromRing(Partitioner partitioner,
         return new CassandraRing(partitioner, keyspace, replicationFactor, instances);
     }
 
+    // Startup Validation
+
+    @Override
+    public void startupValidate()
+    {
+        StartupValidator.instance().register(new SidecarValidation(sidecar));
+        StartupValidator.instance().register(new CassandraValidation(sidecar));
+        StartupValidator.instance().perform();
+    }
```

Comment:
Same comment here. It feel to me that the interface validate-able is tangling/clashing with the validator/validation classes.

Reply:
Let me clarify. 

The register and perform design is unnecessary, as commented in https://github.com/apache/cassandra-analytics/pull/15#discussion_r1330931869

I agree that the validation should be performed eagerly. 

In the code, it looks like this. 

```java 
{
    new SidecarValidation(sidecar).validate();
    new CassandraValidation(sidecar).validate();
}
```

Reply:
(See my reply to the previous comment.)